File: Session11/C++/SymmetricTree.cpp
https://leetcode.com/problems/symmetric-tree/

class Solution {
    
    bool isSymm(TreeNode* left, TreeNode* right)
    {
        if(left==NULL || right==NULL) {
            return left==right;
        }
        else if(left->val!=right->val) {
            return false;
        }
        return isSymm(left->left, right->right) && isSymm(left->right, right->left);
    }
    
public:
    bool isSymmetric(TreeNode* root) {
        
        return isSymm(root->left, root->right);   
    }
};

File: Session11/Java/SymmetricTree.java
https://leetcode.com/problems/symmetric-tree/

class Solution {
    
    boolean symm(TreeNode left, TreeNode right)
    {
        if(left==null || right==null) {
            return left==right;
        }
        else if(left.val!=right.val) {
            return false;
        }
        return symm(left.left, right.right) && symm(left.right, right.left);
    }
    
    public boolean isSymmetric(TreeNode root) {
        
        if(root == null)
        {
            return true;
        }
        
        return symm(root.left, root.right);
    }
}